eth_client/test/eth_rpc_test.exs

```diff
@@ -0,0 +1,99 @@
+defmodule EthClientTest.Rpc do
+  use ExUnit.Case
+  alias EthClient
+  alias EthClient.Context
+  alias EthClient.Rpc
+
+  @bin_path "../contracts/src/bin/Storage.bin"
+
+  defp add_0x(data), do: "0x" <> data
+
+  def transaction_deploy do
+    :ok = EthClient.set_chain("local")
+    {:ok, data} = File.read(@bin_path)
+    data = add_0x(data)
+
+    caller = Context.user_account()
+    caller_address = String.downcase(caller.address)
+    contract_address = Context.contract().address
+
+    nonce = EthClient.nonce(caller.address)
+    gas_limit = EthClient.gas_limit(data, caller_address)
+
+    raw_tx =
+      EthClient.build_raw_tx(
+        0,
+        nonce,
+        gas_limit,
+        EthClient.gas_price(),
+        data: data
+      )
+
+    tx_hash = EthClient.sign_transaction(raw_tx, caller.private_key)
+
+    %{
+      data: data,
+      caller: caller,
+      caller_address: caller_address,
+      contract_address: contract_address,
+      nonce: nonce,
+      gas_limit: gas_limit,
+      raw_tx: raw_tx,
+      tx_hash: tx_hash
+    }
+  end
+
+  setup_all do
+    transaction_deploy()
```

Comment:
i think that this can be achieved using "EthClient.deploy(bin_path)". as all the variables should have been setted up in env variables, there should not be a need to set them by hand. You can also use 'EthClient,Context.all()" to get all the context configurations. We can discuss this if it is needed.